Source/Core/Core/IOS/USB/Emulated/Microphone-Logitech.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2025 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#pragma once
+
+#include "Common/CommonTypes.h"
+#include "Core/IOS/USB/Emulated/Microphone.h"
+
+namespace IOS::HLE::USB
+{
+class LogitechMicState;
+
+class MicrophoneLogitech final : public Microphone
```

Comment:
I see the reason for the this separate class, but I think the current naming of the source files makes things really hard to navigate.

This class more or less just configures the host microphone class behavior a bit.

If you change `m_microphone` in `LogitechMic` from`std::unique_ptr<MicrophoneLogitech>` to
`std::unique_ptr<Microphone>` then "LogitechMic.cpp" is the only place where `MicrophoneLogitech` is needed.

The definition of `MicrophoneLogitech` can then be moved to the top of "LogitechMic.cpp" with all the member functions defined there inside the class body.

This same can be done with `MicrophoneWiiSpeak` and  "WiiSpeak.cpp".

This will make things far less verbose, eliminate the four "Microphone-*" files, and I think make things a lot easier to navigate.